src/board.rs

```diff
@@ -131,14 +144,23 @@ impl Board {
             }
         }
 
-        for y in 0..self.height() {
-            for x in 0..self.width() {
+        let rows = self.height();
+        let cols = self.width();
+        let squares = (0..rows).flat_map(|y| (0..cols).zip(std::iter::repeat(y)));
```

Comment:
Useful func: [Iterator::zip](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip)

Zips iterators A and B into one iterator of `(A.0, B.0), (A.1, B.1), ...` — concluding when either iterator finishes

Reply:
Used here instead of the nested for because it lets us map the `(x,y)` iterator into changes tidily